### ISSUE TITLE
How to use category feature  in xgboost model？

### ISSUE THREAD

**Issue by poson** [o19s/elasticsearch-learning-to-rank#160]
In my xgboost model，we have a  list  of query_id feature，query id has been convert to one-hot feature as：query_1,query_2 ,query_3 。so how can we use elasticsearch-learning-to-rank to run this xgboost model?

**Comment by nomoa** [comment]
Could you clarify your question? Is `query_1` a feature that is determined outside the plugin or something that depends on the data in the index?
If your categories are purely query dependent perhaps the way to do this would be to create a feature per category such as :
```
[
  {
     "name" : "query_1",
     "template" : "q1",
     "template_language" : "derived_expression",
     "params" : [ "q1" ]
  },
  {
    "name" : "query_2",
    "template" : "q2",
    "template_language" : "derived_expression",
    "params" : [ "q2" ]
  }
]
```

Then when submitting the `sltr` query you would pass the parameters `q1` and `q2` : 
```
{ "sltr" : {
   "model" : "my_model",
   "params": {
         "q1" : 1,
         "q2" : 0
   }
}
```
Will emit a feature vector like this to the ranker
* `query_1`: 1
* `query_2`: 0
 
This is using a new feature available only in recent versions of the plugin which allows to pass query time arguments to derived_expressions.
Basically  you'll use the `sltr` query params to pass the query dependent features.
There should be perhaps a more straightforward way to do this.

**Comment by softwaredoug** [comment]
@poson did @nomoa's suggestion make sense?

**Comment by zhuqingrui** [comment]
> Could you clarify your question? Is `query_1` a feature that is determined outside the plugin or something that depends on the data in the index?
> If your categories are purely query dependent perhaps the way to do this would be to create a feature per category such as :
> 
> ```
> [
>   {
>      "name" : "query_1",
>      "template" : "q1",
>      "template_language" : "derived_expression",
>      "params" : [ "q1" ]
>   },
>   {
>     "name" : "query_2",
>     "template" : "q2",
>     "template_language" : "derived_expression",
>     "params" : [ "q2" ]
>   }
> ]
> ```
> Then when submitting the `sltr` query you would pass the parameters `q1` and `q2` :
> 
> ```
> { "sltr" : {
>    "model" : "my_model",
>    "params": {
>          "q1" : 1,
>          "q2" : 0
>    }
> }
> ```
> Will emit a feature vector like this to the ranker
> 
> * `query_1`: 1
> * `query_2`: 0
> 
> This is using a new feature available only in recent versions of the plugin which allows to pass query time arguments to derived_expressions.
> Basically you'll use the `sltr` query params to pass the query dependent features.
> There should be perhaps a more straightforward way to do this.

=====......in recent versions?
not for ltr-es.6.2.2?
